### PR TITLE
feat: Add loading state for all kinds of components

### PR DESCRIPTION
**Pull request by killbond007**
## Description

Add loading state for all kinds of components

Capture d’écran 2021-03-22 à 09 25 55
Capture d’écran 2021-03-22 à 09 22 33
Capture d’écran 2021-03-22 à 09 21 49

## Links

- https://github.com/stargately/launch-room/issues/75

### CODE REVIEW

**Review by puncsky — APPROVED** [review]

File: src/shared/feature-flags/flags-status-table.tsx [hunk 1]
```diff
@@ -65,21 +67,25 @@ export const FlagsStatusTable: React.FC<Props> = ({ data }) => {
...
-              {t("feature_flags.not_found_title")}
-            </Typography.Title>
-            <Typography.Text type="secondary">
-              {t("feature_flags.not_found_description")}
-            </Typography.Text>
-            <NewFlagController newFlagLabel="CREATE FLAG" />
-          </Space>
-        )}
+        renderEmpty={() =>
+          isLoading ? (
```
Comment (bitQ2019) [inline src/shared/feature-flags/flags-status-table.tsx:71]: You can wrap Space directly using Skeleton.
Like https://codesandbox.io/s/w2k24

File: src/shared/flag-details/rules.tsx [hunk 2]
```diff
@@ -10,28 +10,34 @@ import Form from "antd/lib/form";
...
           <Space direction="vertical" style={{ width: "100%" }}>
-            {fields.map((field) => (
-              <Rule
-                variance={variance}
-                index={field.key}
-                key={field.key}
-                remove={() => remove(field.name)}
-              />
-            ))}
+            {isLoading ? (
```
Comment (bitQ2019) [inline src/shared/flag-details/rules.tsx:29]: same as above

File: src/shared/feature-flags/flags-status-table-controller.tsx [hunk 3]
```diff
@@ -9,15 +9,15 @@ export const FlagsStatusTableController: React.FC = () => {
...
+  const { flagsStatus, refetch, loading } = useFlagsStatus({
     workspaceId,
     skip: 0,
     limit: 10000,
   });
   return (
     <WorkspaceIdContext.Provider value={workspaceId}>
       <RefetchContext.Provider value={refetch}>
-        <FlagsStatusTable data={flagsStatus?.flags || []} />;
+        <FlagsStatusTable data={flagsStatus?.flags || []} isLoading={loading} />
```
Comment (bitQ2019) [inline src/shared/feature-flags/flags-status-table-controller.tsx:20]: I recently found that isLoading === loading. 
They just use loading...

**Review by bitQ2019 — APPROVED** [review]

File: src/shared/flag-details/rules.tsx (hunk 2 above)
Comment (killbond007) [reply src/shared/flag-details/rules.tsx]: thx :)

**Pull request merged**